utils/v2tofhirr4/mappings.bal

```diff
@@ -36,17 +37,27 @@ public function stringToHl7(string msg) returns hl7:Message|error {
 
 # Transform an HL7 message to FHIR.
 #
-# + hl7 - HL7 message as a string or an hl7v2:Message
+# + hl7 - HL7 message as a string or an hl7v2:Message  
+# + customMapper - Custom mapper implementation
 # + return - FHIR Bundle as a json
-public function v2ToFhir(string|hl7:Message hl7) returns json|error {
+public function v2ToFhir(string|hl7:Message hl7, SegmentToFhir? customMapper = ()) returns json|error {
     hl7:Message hl7msg;
     if (hl7 is string) {
         hl7msg = check stringToHl7(hl7);
     } else {
         hl7msg = hl7;
     }
-    json transformToFHIRResult = check transformToFhir(hl7msg);
-    return transformToFHIRResult;
+    SegmentToFhir mapperImpl = getMapperContext().getDefaultImpl();
+    if customMapper is () {
+        return check transformToFhir(hl7msg, mapperImpl);
+    }
+    foreach string key in customMapper.keys() {
+        if customMapper.get(key) != () {
+            //binding the custom mapping functions
+            mapperImpl[key] = customMapper.get(key);
+        }
+    }
+    return check transformToFhir(hl7msg, mapperImpl);
```

Comment:
```suggestion
    return transformToFhir(hl7msg, mapperImpl);
```

`check` will do an unnecessary check since you are returning anyway.

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerinax-health.hl7v2/pull/35